### ISSUE TITLE
logical refinements: type alias + Refine doesn't have info from other type aliases

### ISSUE THREAD

**Issue by AlexKnauth** [racket/typed-racket#641]
#### What version of Racket are you using?

Version 6.11.0.2--2017-11-03 (4f1ef42d07/a)

#### What program did you run?

```racket
#lang typed/racket/base #:with-refinements

(define-type Lyst (List Integer))

(define-type Lyst5
  (Refine [l : Lyst] (= (car l) 5)))
```

#### If you got an error message, please include it here.

```
=: terms in linear constraints must be integers, got Any for (car l)
  parsing context: 
   while parsing symbolic object in an inequality
   while parsing proposition in: (car l)
```

#### What should have happened?

It should have typechecked, and behaved the same way as if I manually substituted the `Lyst` type alias:
```racket
(define-type Lyst5
  (Refine [l : (List Integer)] (= (car l) 5)))
```

**Comment by pnwamk** [comment]
thanks for the report!

(surprised this didn't just work -- hopefully a trivial fix)

**Comment by AlexKnauth** [comment]
Fixed by #643.
